zebra-network/src/peer_set/set/tests/prop.rs

```diff
@@ -90,17 +94,85 @@ proptest! {
             Ok::<_, TestCaseError>(())
         })?;
     }
+
+    /// Test if requests are broadcasted to the right number of peers.
+    #[test]
+    fn broadcast_to_peers(
+        total_number_of_peers in (1..100usize)
+    ) {
+        // Get a dummy block hash to help us construct a valid request to be broadcasted
+        let block: block::Block = zebra_test::vectors::BLOCK_MAINNET_10_BYTES
+            .zcash_deserialize_into()
+            .unwrap();
+        let block_hash = block::Hash::from(&block);
+
+        // Start the runtime
+        let runtime = zebra_test::init_async();
+        let _guard = runtime.enter();
+
+        let peer_versions = vec![CURRENT_NETWORK_PROTOCOL_VERSION; total_number_of_peers];
+        let peer_versions = PeerVersions {
+            peer_versions,
+        };
+
+        // Get peers and handles
+        let (discovered_peers, mut handles) = peer_versions.mock_peer_discovery();
+        let (minimum_peer_version, _best_tip_height) =
+            MinimumPeerVersion::with_mock_chain_tip(Network::Mainnet);
+
+        // Build a peerset
+        runtime.block_on(async move {
+            let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
+                .with_discover(discovered_peers)
+                .with_minimum_peer_version(minimum_peer_version.clone())
+                .build();
+
+            // Get the total number of active peers
+            let total_number_of_active_peers = check_if_only_up_to_date_peers_are_live(
+                &mut peer_set,
+                &mut handles,
+                CURRENT_NETWORK_PROTOCOL_VERSION,
+            )?;
+
+            // The number of active peers is never greater than the total
+            prop_assert!(total_number_of_peers >= total_number_of_active_peers);
```

Comment:
I think that all peers should be active?
```suggestion
            // Since peer addresses are unique, and versions are valid, every peer should be active
            prop_assert_eq!(total_number_of_peers, total_number_of_active_peers);
```

Reply:
Not all the peers are active, this change will probably make the test fail. I can recheck this tomorrow but i am pretty sure this is the case.

Reply:
This could be a bug, either in this test, the test harness, or the `PeerSet`.

Can you try to find out why some peers are not active?

Reply:
will try tomorrow.

Reply:
Apologies, it seems that i had this difference this with some other version of the test or something because i tried several times now and it is always the same so the `assert_eq!` check here is fine. I applied the changes at https://github.com/ZcashFoundation/zebra/pull/3284/commits/53445d631884e48c26a8e2345e5782c0bd14b47d

Reply:
Yeah I think the original code dropped some peers because their versions were too low. But now their versions are all correct.